src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useContext, useEffect, useState } from "react";
+import DatePicker, { registerLocale } from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { Input } from "semantic-ui-react";
+import { AppContext } from "../../../../context";
+import { FilterType, PageState, State } from '../../../../types';
+import { updateFilters } from "../../../../utils/stateUpdateUtils";
+import Translate from "../../../../utils/translate/translateService";
+import SectionHeader from "../SectionHeader";
+import {enUS, fr} from 'date-fns/locale'
+//TODO: Is this supposed to be used
+import { sendFiltersAnalyticsEvent } from "../../../../utils/analyticsUtils";
+import NewDateSlider from '../new-datepicker/NewDateSlider';
+import '../datepicker/DateSlider.css'
+
+interface DatepickerProps {
+  page: string
+}
+
+export default function NewDatepicker({ page }: DatepickerProps) {
+
+  const [state, dispatch] = useContext(AppContext);
+  const filters = (state[page as keyof State] as PageState).filters;
+  const earliestPublicationDate = state.calendarStartDates.minDate;
+  const latestPublicationDate = state.calendarStartDates.maxDate;
+  const millisecondsPerDay = 86400000;
+  const [sliderRange, setSliderRange] = useState<number>((latestPublicationDate.getTime()-earliestPublicationDate.getTime())/millisecondsPerDay); //in days
+  const [filterStartDate, filterEndDate]: Date[] = Array.from(filters.publish_date);
+
+  // TODO: should these functions be in util since one of them is used in both this file and the slider file?
+  const fromDateToNumber = (date: Date) => {
+    return (date.getTime()-earliestPublicationDate.getTime())/millisecondsPerDay
+  }
+  const toDateSinceMinDate = (val: number) => {
+    return new Date(earliestPublicationDate.getTime() + (val*millisecondsPerDay));
+  }
+
+  //dates as number of days since mindate
+  const [chosenDates, setChosenDates] = useState<number[]>([fromDateToNumber(filterStartDate), fromDateToNumber(filterEndDate)])
+
+  useEffect(() => {
+    registerLocale("en", enUS)
```

Comment:
Reminder: will need to register de after the german PR is merged

Reply:
Maybe we should wait for German PR to get merged, and then fix it once and for all in this PR?

@annoyingpr @atatmaja

Reply:
updated!